shell/browser/native_window_mac.mm

```diff
@@ -1810,6 +1812,27 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
   [window_ setAcceptsMouseMovedEvents:forward];
 }
 
+gfx::Rect NativeWindowMac::GetWindowControlsOverlayRect() {
+  gfx::Rect bounding_rect;
+  if (!has_frame() && buttons_view_ && ![buttons_view_ isHidden] &&
+      !traffic_light_position_) {
+    NSRect button_frame = [buttons_view_ frame];
+    const int overlay_width = GetContentSize().width() - NSWidth(button_frame);
+    int overlay_height = NSHeight(button_frame) + 6;
+    if (title_bar_style_ == TitleBarStyle::kHiddenInset) {
+      overlay_height += 15;
```

Comment:
The calculation should not involve hardcoded numbers, we should be able to calculate with existing margins:

```obj-c
NSRect button_frame = [buttons_view_ frame];
CGFloat overlay_height = button_frame.frame.height + button_frame.origin.y * 2;
```